sklearn/compose/_column_transformer.py

```diff
@@ -622,10 +656,10 @@ def _validate_output(self, result):
             name for name, _, _, _ in self._iter(fitted=True, replace_strings=True)
         ]
         for Xs, name in zip(result, names):
-            if not getattr(Xs, "ndim", 0) == 2:
+            if not getattr(Xs, "ndim", 0) == 2 and not hasattr(Xs, "__dataframe__"):
                 raise ValueError(
                     "The output of the '{0}' transformer should be 2D (scipy "
-                    "matrix, array, or pandas DataFrame).".format(name)
+                    "matrix, array, or DataFrames).".format(name)
```

Comment:
I am okay with changing the order. Looking at it again, it should say like "array, sparse matrix, or dataframe".

Reply:
Still not addressed.

Reply:
or maybe "numpy array, scipy sparse array, dataframe" since in the (semi distant) future scipy sparse matrices will all be deprecated in favor of scipy sparse arrays...